tcms/testplans/models.py

```diff
@@ -94,15 +94,22 @@ def list(cls, query=None):
     def confirmed_case(self):
         return self.case.filter(case_status__name='CONFIRMED')
 
-    def add_case(self, case, sortkey=0):
+    def add_case(self, case, sortkey=None):
 
-        tcp, is_created = TestCasePlan.objects.get_or_create(
+        if sortkey is None:
+            lastcase = self.testcaseplan_set.order_by('-sortkey').first()
+            if lastcase and lastcase.sortkey is not None:
+                sortkey = lastcase.sortkey + 10
+            else:
+                sortkey = 0
+
+        return TestCasePlan.objects.get_or_create(
             plan=self,
             case=case,
-        )
-        if is_created:
-            tcp.sortkey = sortkey
-            tcp.save()
+            defaults={
+                'sortkey': sortkey
+            }
```

Comment:
Why not just `sortkey=sortkey` ?

Reply:
The `defaults` parameter is a dictionary that specifies the extra values that are used when creating the object if it doesn't exist yet. To check if there is already an object it uses all the parameters except the `defaults` parameter. So in this case it will look for a `TestCasePlan` filtering on `plan` and `case` and if it finds that, that object is returned. If it doesn't find an object, it will create one and will set `plan`, `case` and also `sortkey`. This way you don't need to do this after the `get_or_create()` call:

```
 if is_created:
     tcp.sortkey = sortkey
     tcp.save()
```

So the functionality is the same, except that it is done in 1 query instead of 2.

Reply:
Cool, I didn't know that.